app/models/user_balance.rb

```diff
@@ -32,6 +32,6 @@ def self.debts_to(user)
   end
 
   def self.debtors_ids(user)
-    user.balances_as_payer.select('user_id').uniq.map(&:user_id)
+    user.balances_as_payer.select('user_id').distinct.map(&:user_id)
```

Comment:
It was about replacing `#map` with `#pluck`  but with your last comment, it doesn't matter anymore.